packages/react-native-audio-api/src/system/AudioManager.ts

```diff
@@ -42,10 +32,14 @@ class AudioManager {
     AudioManagerModule.observeAudioInterruptions(enabled);
   }
 
-  enableRemoteCommand(
-    name: RemoteControlEventName | InterruptionEventName,
+  observeVolumeChanges(enabled: boolean) {
+    AudioManagerModule.observeVolumeChanges(enabled);
+  }
+
+  enableRemoteCommand<Name extends RemoteEventName>(
+    name: Name,
     enabled: boolean,
-    callback?: (event: AudioManagerEventType) => void
+    callback?: RemoteEventCallback<Name>
   ): EmitterSubscription | null {
```

Comment:
I would split this function into two: f.e. `enableRemoteCommand` and `enableRemoteEvent` as it the `remote-command` might be misleading with events like interruption or volume change.

If we wan't to stay with only one method for adding event handlers I would go with `addEventListener`.

Also It would be great to get rid of the `enabled` parameter

i.e.

```
useEffect(() => {
  const listener = AudioManager.addEventListener('onVolumeChange', (volume) => {
    // do stuff
  });
  
  return () => {
    listener.remove();
  }
}, [deps]);
```